actionview/lib/action_view/helpers/cache_helper.rb

```diff
@@ -198,10 +198,12 @@ def fragment_cache_key(key)
 
     private
 
-      def fragment_name_with_digest(name) #:nodoc:
-        if @virtual_path
+      def fragment_name_with_digest(name, virtual_path: nil) #:nodoc:
+        virtual_path ||= @virtual_path
+        if virtual_path
           names  = Array(name.is_a?(Hash) ? controller.url_for(name).split("://").last : name)
-          digest = Digestor.digest name: @virtual_path, finder: lookup_context, dependencies: view_cache_dependencies
+          dependencies = defined?(view_cache_dependencies) ? view_cache_dependencies : []
```

Comment:
@kaspth I was having issues with the tests on `CachedCollectionViewRenderTest`. They use `ActionView::Base` and `view_cache_dependencies` was not defined. The same will not happen if you run `AutomaticCollectionCacheTest`.

I'd rather not have to check for `view_cache_dependencies ` at all if possible. Any ideas? Thanks reviewing!